### PR TITLE
Implementing timestamps

### PR DESCRIPTION
**Pull request by schoettl**
I started with a (hopefully robust) skeleton of BNF rules for parsing timestamps. Co-workers are welcome.

I don't even know yet how to lint / syntax check / run tests – but I plan to further work on this.

- [x] find and cite orgmode specification on timestamps
- [x] implement day part
- [x] implement "modifiers" i.e. repeaters
- [x] check what's missing
- [x] check on how it's about leading zeros in dates/times
- [ ] check the precision of time (fractional seconds?)
- [x] add code (only BNF rules are not enough, e.g. to validate dates/times or to convert to higher-level objects) – OUT OF SCOPE of this PR

### CODE REVIEW

**Comment by munen** [comment]
@schoettl Awesome that you're pushing your knowledge from the organice parser into the new org-parser! :pray: :bowing_man: 

I have added a little bit of doc so new contributors can see how to run tests: https://github.com/200ok-ch/org-parser#prerequisites

If you have any questions getting Clojure to run on your system, feel free to ping us any time!

**Comment by schoettl** [comment]
I think we can merge this branch.

- It does not accept diary timestamps `<%%(exp)>`, but we can add this later.
- It does support seconds although they are discarded in orgmode on C-c C-c. However, they can be parsed easily and they are no reason to let the parser fail.
- I don't want to implement fractional seconds for times. Never mentioned in orgmode docs and hardly used out there.
- As discussed [here](https://github.com/200ok-ch/organice/issues/255), I currently don't want to implement 12-hour-clock format as I think it's not officially supported by orgmode.

Apart from that, I hope I've got all features of timestamps.

Transformation to an easier reusable and higher-level format is out of scope of this PR.

File: resources/org.ebnf [hunk 1]
```diff
@@ -72,4 +72,71 @@ keyword-value = anything-but-newline
...
+
+(* a time span *)
+ts-inner-span = ts-inner-w-time <'-'> ts-time ts-modifiers
+
+ts-inner-w-time  = ts-date [<' '+> ts-day] <' '+> ts-time
+ts-inner-wo-time = ts-date [<' '+> ts-day]
+
+ts-date = #"\d{4}-\d{2}-\d{2}"
+
+ts-time = #"\d{1-2}:\d{2}(:\d{2})?"
```
Comment (munen) [inline resources/org.ebnf]: @schoettl Amazing job as always on this PR! :pray: :bow: 

Just one question, should this line not be: `ts-time = #"\d{1,2}:\d{2}(:\d{2})?" `

If I make this change, the repl starts and the tests will run, if I don't make the change, the REPL crashes with https://gist.github.com/munen/8f701b56c57490a47f7257b7136a070f

**Review by munen — CHANGES_REQUESTED** [review]

File: resources/org.ebnf (hunk 1 above)
Comment (schoettl) [reply resources/org.ebnf]: Oh right, I fixed this already, but as a magit beginner, I must have accidently discarded this...

Comment (schoettl) [reply resources/org.ebnf]: Now, the all tests should run.

**Comment by munen** [comment]
Superb PR, as always, @schoettl. Happy to merge it in, great work! :bow:

**Pull request merged**